src/submission/submission-updates-messenger.ts

```diff
@@ -21,48 +21,63 @@ class DummyMessenger implements SubmissionUpdatesMessenger {
 }
 
 class KafkaMessenger implements SubmissionUpdatesMessenger {
-  private producer: Producer;
-  private knownTopics: ExpectedTopicsMap;
+  private kafka: Kafka;
+  private programUpdateTopic: TopicConfig;
 
-  constructor(clientId: string, brokers: string[], topics: ExpectedTopicsMap) {
-    this.producer = new Kafka({ clientId, brokers }).producer();
-    this.knownTopics = topics;
+  constructor(clientId: string, brokers: string[], topics: TopicConfig) {
+    this.kafka = new Kafka({ clientId, brokers });
+    this.programUpdateTopic = topics;
   }
 
   sendProgramUpdatedMessage = async (programId: string) => {
     L.info('KafkaMessenger called to send message to broker for updated program: ' + programId);
-    await this.producer.connect().catch(this.producerErrorHandler);
-    await this.producer
+
+    await this.createMissingProgramUpdateTopic().catch(this.errorHandler);
```

Comment:
why don't we just do this in the constructor? the topic just needs to be created once, it just adds extra overhead if we do this for every event

Reply:
also once the topic is created, admin connection should be closed.

Reply:
yea Bashar said the same thing, it makes sense to do it that way will change :+1: